includes/modules/shipping/usps.php

```diff
@@ -367,7 +367,7 @@ protected function storefrontInitialization()
         // Quick return if the shipping-module's configuration will not allow it to
         // gather valid USPS quotes.  The shipping-module is disabled if this is the case.
         //
-        if ($this->checkConfiguration() === false) {
+        if ($this->checkConfiguration() === false || $this->checkCartContents() === false) {
```

Comment:
Why not just add the notification to the `checkConfiguration` method.  It seems superfluous to have a method that simply returns an observer's response to a notification.

Reply:
Could we please just merge this?  I don't want to do another round of testing.  I really need to wrap this up.  These are two functions that do two different things.  Who knows the checkCartContents() function could easily grow over time.

Reply:
My point is that where the `checkCartContents` is called is during initialization and doesn't address a site that wants to limit (as opposed to totally disallow, which is what your change does).

I'll merge it, but expect to see it changed to where I think the check _should_ be post-merge.